### PR TITLE
Fix #5635

### LINKED ISSUE #5635: docfx crash report: System.InvalidOperationException
# docfx crash report: System.InvalidOperationException

docfx: `1.0.0+fbb9d2d585952542e6940371b77f885ace71831f`
dotnet: `3.1.200`
x64: `True`
git: `git version 2.21.0.windows.1`
DOCFX_REPOSITORY_URL: `https://github.com/MicrosoftDocs/OfficeDocs-OfficeUpdates-pr`
DOCFX_GLOBAL_CONFIG_PATH: `C:\Users\jipe\Desktop\docfx.global.json`

## repro steps

Run `E:\docfx\src\docfx\bin\Debug\netcoreapp3.1\docfx.dll build D:\docs-repos\OfficeDocs-OfficeUpdates-pr --legacy` in `E:\docfx\src\docfx\bin\Debug\netcoreapp3.1`

## callstack

```
System.InvalidOperationException: Error reading 'common/Office/media/icons/Small-business.svg [common]'
   at Microsoft.Docs.Build.Input.ReadStream(FilePath file) in E:\docfx\src\docfx\build\context\Input.cs:line 136
   at Microsoft.Docs.Build.Output.Copy(Document file, String destRelativePath) in E:\docfx\src\docfx\build\context\Output.cs:line 62
   at Microsoft.Docs.Build.BuildResource.Build(Context context, Document file) in E:\docfx\src\docfx\build\resource\BuildResource.cs:line 43
   at Microsoft.Docs.Build.Build.BuildFile(Context context, FilePath path) in E:\docfx\src\docfx\build\Build.cs:line 150
   at Microsoft.Docs.Build.Build.Run(Config config, Docset docset, Docset fallbackDocset, CommandLineOptions options, ErrorLog errorLog, String outputPath, Input input, RepositoryProvider repositoryProvider, LocalizationProvider localizationProvider, PackageResolver packageResolver) in E:\docfx\src\docfx\build\Build.cs:line 93
   at Microsoft.Docs.Build.Build.BuildDocset(String docsetPath, String outputPath, CommandLineOptions options) in E:\docfx\src\docfx\build\Build.cs:line 60
   at Microsoft.Docs.Build.Build.Run(String workingDirectory, CommandLineOptions options) in E:\docfx\src\docfx\build\Build.cs:line 25
   at Microsoft.Docs.Build.Docfx.Run(String[] args) in E:\docfx\src\docfx\cli\Docfx.cs:line 82
   at Microsoft.Docs.Build.Docfx.Main(String[] args) in E:\docfx\src\docfx\cli\Docfx.cs:line 26
```

### PR DESCRIPTION
**Pull request by yufeih**
Fixes dependency file handling #5635 

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/5636)

### CODE REVIEW

**Comment by live1206** [comment]
Add the spec test that docfx should not read from git commit history for dependency files?

**Comment by yufeih** [comment]
> Add the spec test that docfx should not read from git commit history for dependency files?

Cannot test it with spec test because there is no observable side effect. The content of checkout file is the same as the content in HEAD :(

But let me try to repro this using the actual repo tomorrow when I'm in office.

**Comment by yufeih** [comment]
The diff is caused by extra `\r` for code blocks.

File: src/docfx/restore/PackageResolver.cs [hunk 1]
```diff
@@ -138,6 +138,7 @@ private void DownloadGitRepositoryCore(string cwd, string url, string committish
 
             Directory.CreateDirectory(cwd);
             GitUtility.Init(cwd);
+            GitUtility.AddRemote(cwd, "origin", url);
```
Comment (yufeih) [inline src/docfx/restore/PackageResolver.cs]: Now that our repository is derived from disk state, ensure repository url here.

File: src/docfx/build/context/Input.cs [hunk 2]
```diff
@@ -146,60 +137,103 @@ public FilePath[] ListFilesRecursive(FileOrigin origin, PathString? dependencyNa
             switch (origin)
             {
                 case FileOrigin.Default:
```
Comment (yufeih) [inline src/docfx/build/context/Input.cs]: Merge Input and repository provider as they are pretty intertwined.

**Comment by yufeih** [comment]
The reason for the `\r` diff is that we now read from dependency files that are checked out, vs previously it is read from git commit blob.

File: src/docfx/build/context/Input.cs [hunk 3]
```diff
@@ -146,60 +150,54 @@ public FilePath[] ListFilesRecursive(FileOrigin origin, PathString? dependencyNa
...
-                    .ToArray();
+                return Array.Empty<string>();
             }
 
-            FilePath CreateFilePath(string path)
-            {
-                return dependencyName is null ? new FilePath(path, origin) : new FilePath(path, dependencyName.Value);
-            }
+            return from file in Directory.GetFiles(directory, "*", SearchOption.AllDirectories)
+                   where !file.Contains(".git/") && !file.Contains(".git\\")
```
Comment (live1206) [inline src/docfx/build/context/Input.cs]: why can not it just be `.git`?

Comment (yufeih) [reply src/docfx/build/context/Input.cs]: a file may be named `contain.git.md`

File: src/docfx/lib/path/FilePath.cs [hunk 4]
```diff
@@ -44,32 +39,20 @@ public FilePath(string path, FileOrigin origin = FileOrigin.Default)
             Origin = origin;
         }
 
-        public FilePath(string path, string? commit, FileOrigin origin)
+        public FilePath(string path, bool isGitCommit)
         {
             Path = new PathString(path);
-            Origin = origin;
-            Commit = commit;
+            Origin = FileOrigin.Fallback;
```
[the remark this answers is not in the document]
Comment (yufeih) [reply src/docfx/lib/path/FilePath.cs]: test will fail?

**Pull request merged**